bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerDeviceHandler.java

```diff
@@ -116,6 +116,19 @@ protected void updateChannel(ChannelUID channel) {
                 break;
             case CHANNEL_RESET_STATIONS:
                 break;
+            case CHANNEL_QUEUED_ZONES:
+                updateState(channel, new DecimalType(localAPI.getQueuedZones()));
+                break;
+            case CHANNEL_CLOUD_CONNECTED:
+                if (localAPI.getCloudConnected() == 3) {
+                    updateState(channel, OnOffType.ON);
+                } else {
+                    updateState(channel, OnOffType.OFF);
+                }
```

Comment:
Just a suggestion: it can be reduced to:
```
updateState(channel, OnOffType.valueOf(localAPI.getCloudConnected() == 3));
```

Reply:
I tried that but "OnOffType.valueOf(" does not support a boolean. It would be a good idea to extend/overload OnOffType.valueOf to support doing that.

Reply:
public static [OnOffType](https://www.openhab.org/javadoc/latest/org/openhab/core/library/types/onofftype) from(boolean state)

Used your suggestion but with .from(boolean)

Reply:
Sorry for mentioning the wrong method.